packages/use-dataloader/src/DataLoaderProvider.js

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types'
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+} from 'react'
+
+export const DataLoaderContext = createContext()
+
+const DataLoaderProvider = ({ children }) => {
+  const cachedData = useRef({})
+  const reloads = useRef({})
+
+  const setCachedData = useCallback(compute => {
+    if (typeof compute === 'function') {
+      cachedData.current = compute(cachedData.current)
+    } else {
+      cachedData.current = compute
+    }
+  }, [])
+
+  const setReloads = useCallback(compute => {
+    if (typeof compute === 'function') {
+      reloads.current = compute(reloads.current)
+    } else {
+      reloads.current = compute
+    }
+  }, [])
+
+  const addCachedData = useCallback(
+    (key, newData) => {
+      if (key && typeof key === 'string' && newData) {
+        setCachedData(actualCachedData => ({
+          ...actualCachedData,
+          [key]: newData,
+        }))
+      }
+    },
+    [setCachedData],
+  )
+
+  const addReload = useCallback(
+    (key, method) => {
+      if (key && typeof key === 'string' && method) {
+        setReloads(actualReloads => ({
+          ...actualReloads,
+          [key]: method,
+        }))
+      }
+    },
+    [setReloads],
+  )
+
+  const clearReload = useCallback(
+    key => {
+      if (key && typeof key === 'string') {
+        setReloads(actualReloads => {
+          const tmp = actualReloads
+          delete tmp[key]
+
+          return tmp
+        })
+      } else {
+        setReloads({})
+      }
+    },
+    [setReloads],
+  )
+
+  const clearCachedData = useCallback(
+    key => {
+      if (key && typeof key === 'string') {
+        setCachedData(actualCachedData => {
+          const tmp = actualCachedData
+          delete tmp[key]
+
+          return tmp
+        })
+      } else {
+        setCachedData({})
+      }
+    },
+    [setCachedData],
+  )
+
+  const reload = useCallback(async key => {
```

Comment:
Reload should be separate in two:
- one only scope to this dataLoader
- one can be used to reload all app ( maybe reloadAllDataLoader ? )